system/jlib/jlog.cpp

```diff
@@ -252,16 +249,16 @@ void LogMsgJobInfo::deserialize(MemoryBuffer & in)
         // >= 8.0 client
         StringBuffer idStr;
         in.read(idStr);
-        jobIDStr = idStr.detach();
+
+        CriticalBlock block(externalJobCS);
```

Comment:
Nervous about adding this atom table without understanding (a) what it is gaining us - is a non-trivial destructor really that painful? and (b) how often this code gets called and will the criticalblock have a worse impact than the original problem

Reply:
I agree - included for discussion